gitlabform/processors/project/protected_environments.py

```diff
@@ -0,0 +1,34 @@
+from gitlabform.gitlab import GitLab
+from gitlabform.processors.defining_keys import Key, And
+from gitlabform.processors.multiple_entities_processor import MultipleEntitiesProcessor
+from gitlabform.processors.util.difference_logger import DifferenceLogger
+
+
+class ProtectedEnvironmentsProcessor(MultipleEntitiesProcessor):
+    """https://docs.gitlab.com/ee/api/protected_environments.html#protect-repository-environments"""
+
+    def __init__(self, gitlab: GitLab):
+        super().__init__(
+            "protected_environments",
+            gitlab,
+            list_method_name=gitlab.list_protected_environments,
+            add_method_name=gitlab.protect_a_repository_environment,
+            edit_method_name=gitlab.update_a_protected_environment,
+            delete_method_name=gitlab.unprotect_environment,
+            # TODO: I had to define "name" inside the cfg, it should get the section's name as the entity name, e.g.
+            # protected_environments:
+            #   foo:
+            #     name: "foo" <- This is redundant
```

Comment:
Yeah, this is not perfect that you have to do this... But I would still stick to it for consistency with other sections.